six/modules/c++/six.sicd/source/SICDWriteControl.cpp

```diff
@@ -103,10 +103,10 @@ void SICDWriteControl::writeHeaders()
         //       SIDD as you will have some pad
         // TODO: Would be nice if ImageSubheader had a method to compute this
         //       for you
-        const size_t numRows(subheader.getNumRows());
-        const size_t numCols(subheader.getNumCols());
-        numBands = subheader.getNumImageBands();
-        const size_t numBitsPerPixel(subheader.getNumBitsPerPixel());
+        const size_t numRows(static_cast<nitf::Uint64>(subheader.getNumRows()));
```

Comment:
Ahh I see... so for both these cases (which is coming from the nitf module's `Field.hpp`) and the below cases (coming from six's `Parameter.h`) the OSX compiler isn't finding a cast operator that's exactly for `size_t` and is complaining.  It's easy enough to add a cast operator for `size_t` to those two classes so that you wouldn't need these casts here, but I'm not sure if that'll compile on OS's where `nitf::Uint64` and `size_t` are the exact same underlying type (rather than just unsigned ints of the same size)... if I remember correctly, they're the exact same type on Windows (I've had annoyances related to this in the past).

If it comes to it, we can live with the casts if we really need to but I'll see if there's another way.